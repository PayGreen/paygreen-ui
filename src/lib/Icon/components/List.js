import React from 'react';

const SvgList = props => (
  <svg viewBox="0 0 143 143" {...props}>
    <path d="M21.9,45.2h-9.7c-1.9,0-3.5-1.6-3.5-3.5s1.6-3.5,3.5-3.5h9.7c1.9,0,3.5,1.6,3.5,3.5S23.8,45.2,21.9,45.2z" />
    <path d="M130.5,45.2H46.6c-1.9,0-3.5-1.6-3.5-3.5s1.6-3.5,3.5-3.5h83.9c1.9,0,3.5,1.6,3.5,3.5S132.4,45.2,130.5,45.2z" />
    <path d="M22,75.6h-9.7c-1.9,0-3.5-1.6-3.5-3.5s1.6-3.5,3.5-3.5H22c1.9,0,3.5,1.6,3.5,3.5S24,75.6,22,75.6z" />
    <path d="M130.7,75.6H46.8c-1.9,0-3.5-1.6-3.5-3.5s1.6-3.5,3.5-3.5h83.9c1.9,0,3.5,1.6,3.5,3.5S132.6,75.6,130.7,75.6z" />
    <path d="M22,105.9h-9.7c-1.9,0-3.5-1.6-3.5-3.5s1.6-3.5,3.5-3.5H22c1.9,0,3.5,1.6,3.5,3.5S24,105.9,22,105.9z" />
    <path d="M130.7,105.9H46.8c-1.9,0-3.5-1.6-3.5-3.5s1.6-3.5,3.5-3.5h83.9c1.9,0,3.5,1.6,3.5,3.5S132.6,105.9,130.7,105.9z" />
  </svg>
);

export default SvgList;
