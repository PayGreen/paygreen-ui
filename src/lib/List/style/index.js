import styled from 'styled-components';
import { responsiveSpaces } from '../../../shared/spaces';

const ListBase = styled.ul`
    list-style-type: none;
    padding: .1px 0;
    ${responsiveSpaces('margin')};
`;

export { ListBase };