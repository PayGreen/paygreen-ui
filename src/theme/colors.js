const mainColors = {
    green: '#33ad73',
    blue: '#00a99d',
    orange: '#e58a28',
    pink: '#ec6a6a',
    purple: '#a26ae1',
    red: '#b54025',
};

const colors = {
    green: {
        main: mainColors.green,
        gradientBase: '#77d1a6',
        gradientShade: '#a4db95',
    },
    blue: {
        main: mainColors.blue,
        gradientBase: '#51c6be',
        gradientShade: '#76bbd8',
    },
    orange: {
        main: mainColors.orange,
        gradientBase: '#fab56b',
        gradientShade: '#faa272',
    },
    pink: {
        main: mainColors.pink,
        gradientBase: '#e98e90',
        gradientShade: '#e98eb9',
    },
    purple: {
        main: mainColors.purple,
        gradientBase: '#b887ee',
        gradientShade: '#dd94ef',
    },
};

export { mainColors, colors };
