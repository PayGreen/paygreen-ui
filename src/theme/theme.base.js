const responsive = {
    sm: 550,
    md: 960,
    lg: 1200,
    xl: 1360,
};

const query = (value, operator) => {
    if (operator == 'max') {
        value -= 1;
    }
    return operator + '-width: ' + value + 'px';
};

const pallet = {
    green: {
        main: '#33ad73',
        light: '#d5f3e5',
    },
    blue: {
        main: '#00a99d',
        light: '#e1f3f2',
    },
    orange: {
        main: '#e58a28',
        light: '#fbe6ce',
    },
};

export const ThemeBase = {
    name: 'Base',
    color: {
        primary: {
            ...pallet.green,
            gradientBase: '#77d1a6',
            gradientShade: '#a4db95',
        },
        secondary: {
            ...pallet.blue,
            gradientBase: '#51c6be',
            gradientShade: '#76bbd8',
        },
        tertiary: {
            ...pallet.orange,
            gradientBase: '#fab56b',
            gradientShade: '#faa272',
        },
        quaternary: {
            main: '#ec6a6a',
            light: '#ffe0e1',
            gradientBase: '#e98e90',
            gradientShade: '#e98eb9',
        },
        quinary: {
            main: '#a26ae1',
            light: '#e9d9fb',
            gradientBase: '#b887ee',
            gradientShade: '#dd94ef',
        },
    },
    status: {
        success: {
            ...pallet.green,
        },
        default: {
            ...pallet.blue,
        },
        warning: {
            ...pallet.orange,
        },
        danger: {
            main: '#b54025',
            light: '#f7d7d0',
        },
    },
    font: {
        size: {
            tiny: '1rem',
            xxs: '1.2rem',
            xs: '1.4rem',
            sm: '1.6rem',
            base: '1.8rem',
            md: '2.1rem',
            lg: '3rem',
            xl: '4rem',
        },
        weight: {
            normal: 400,
            bold: 700,
        },
        lineHeight: {
            base: 1,
            md: 1.5,
        },
        spacing: '0.3em',
        underline: {
            space: {
                tiny: '8px',
                xxs: '10px',
                xs: '10px',
                sm: '12px',
                base: '12px',
                md: '12px',
                lg: '24px',
                xl: '24px',
            },
            line: {
                tiny: '2px',
                xxs: '2px',
                xs: '2px',
                sm: '2px',
                base: '2px',
                md: '2px',
                lg: '3px',
                xl: '3px',
            },
            width: {
                tiny: '30px',
                xxs: '40px',
                xs: '40px',
                sm: '50px',
                base: '50px',
                md: '65px',
                lg: '80px',
                xl: '80px',
            },
        },
    },
    line: '2px',
    dash: '3px',
    radius: {
        none: '0',
        sm: '4px',
        md: '8px',
        lg: '12px',
    },
    arrow: {
        sm: '8px',
        md: '12px',
        lg: '16px',
    },
    space: {
        none: '0',
        xs: '4px',
        sm: '8px',
        md: '16px',
        lg: '32px',
        xl: '64px',
    },
    grid: {
        header: '45px',
        headerBig: '70px',
        topbar: '45px',
        sidebar: '380px',
        maxWidth: '1400px',
    },
    blockShift: {
        none: '0',
        xs: '64px',
        sm: '128px',
        md: '256px',
        lg: '320px',
        xl: '384px',
    },
    blockSpace: {
        none: '0',
        xs: '16px',
        sm: '24px',
        md: '40px',
        lg: '52px',
        xl: '68px',
    },
    blockWidth: {
        sm: '400px',
        md: '560px',
        lg: '640px',
        xl: '800px',
    },
    blockHeader: {
        sm: '140px',
        md: '180px',
        lg: '210px',
        xl: '240px',
    },
    bannerHeight: {
        xs: '33vh',
        sm: '50vh',
        lg: '100vh',
    },
    wave: {
        simple: '8vw',
        double: '11vw',
    },
    icon: {
        size: {
            xxs: '12px',
            xs: '16px',
            sm: '20px',
            md: '24px',
            lg: '32px',
            xl: '40px',
        },
        shift: {
            xxs: '2px',
            xs: '3px',
            sm: '4px',
            md: '4px',
            lg: '6px',
            xl: '6px',
        },
    },
    imageSize: {
        auto: 'auto',
        xxs: '150px',
        xs: '240px',
        sm: '400px',
        md: '560px',
        lg: '640px',
        xl: '760px',
    },
    imageCircleSize: {
        auto: 'auto',
        xxs: '38px',
        xs: '60px',
        sm: '80px',
        md: '100px',
        lg: '120px',
        xl: '140px',
    },
    logoHeight: {
        none: 'auto',
        xs: '20px',
        sm: '30px',
        md: '40px',
        lg: '50px',
        xl: '60px',
    },
    logoWidth: {
        none: 'auto',
        xs: '90px',
        sm: '120px',
        md: '150px',
        lg: '200px',
        xl: '250px',
    },
    loaderSize: {
        xxs: '40px',
        xs: '50px',
        sm: '60px',
        md: '70px',
        lg: '80px',
        xl: '90px',
    },
    transition: {
        xxs: '0.1s',
        xs: '0.2s',
        sm: '0.3s',
        md: '0.5s',
        lg: '0.7s',
        xl: '1s',
    },
    zindex: {
        hidden: -100,
        layer: -5,
        background: -1,
        base: 0,
        content: 1,
        corner: 3,
        front: 5,
        loader: 90,
        subHeader: 100,
        menu: 105,
        header: 110,
        popin: 115,
        modal: 120,
    },
    query: {
        min: {
            sm: query(responsive.sm, 'min'),
            md: query(responsive.md, 'min'),
            lg: query(responsive.lg, 'min'),
            xl: query(responsive.xl, 'min'),
        },
        max: {
            sm: query(responsive.sm, 'max'),
            md: query(responsive.md, 'max'),
            lg: query(responsive.lg, 'max'),
            xl: query(responsive.xl, 'max'),
        },
    },
    button: {
        paddingWidth: {
            sm: '22px',
            md: '26px',
            lg: '30px',
        },
        paddingHeight: {
            sm: '12px',
            md: '14px',
            lg: '16px',
        },
        shift: '3px',
        font: {
            sm: '1.2rem',
            md: '1.4rem',
            lg: '1.5rem',
        },
        letterSpacing: {
            sm: '0.3rem',
            md: '0.4rem',
            lg: '0.45rem',
        },
    },
    daButton: {
        buttonHeight: {
            sm: '32px',
            md: '38px',
            lg: '44px',
        },
        shift: {
            sm: '2px',
            md: '4px',
            lg: '6px',
        },
        font: {
            sm: '1.35rem',
            md: '1.4rem',
            lg: '1.7rem',
        },
        iconSpacing: {
            sm: '0.9rem',
            md: '1rem',
            lg: '1.1rem',
        },
    },
    form: {
        inputWidth: {
            xs: '260px',
            sm: '330px',
            md: '400px',
            lg: '500px',
        },
        textareaHeight: '200px',
        radio: {
            md: '120px',
            lg: '150px',
        },
        checkboxSize: {
            sm: '18px',
            md: '20px',
            lg: '22px',
        },
        radioButtonSize: {
            sm: '19px',
            md: '21px',
            lg: '23px',
        },
        radioButtonSpace: {
            sm: '8px',
            md: '10px',
            lg: '10px',
        },
        toggle: '30px',
    },
    corner: {
        size: {
            lg: '34px',
        },
        padding: {
            lg: '6px',
        },
    },
    shadow: {
        size: {
            none: '0 0 0',
            xs: '1px 1px 5px',
            sm: '1px 1px 10px',
            md: '2px 2px 15px',
            lg: '3px 3px 20px',
        },
        opacity: {
            none: 1,
            xs: 0.95,
            sm: 0.9,
            md: 0.8,
            lg: 0.7,
        },
    },
    rotateSize: {
        d0: '0deg',
        d45: '45deg',
        d90: '90deg',
        d135: '135deg',
        d180: '180deg',
        d225: '225deg',
        d270: '270deg',
        d315: '315deg',
    },
    dataBar: {
        height: '10px',
        width: {
            sm: '300px',
            md: '400px',
            lg: '500px',
            xl: '600px',
        },
    },
};
