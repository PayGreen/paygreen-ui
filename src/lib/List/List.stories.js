import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, radios, select } from '@storybook/addon-knobs';
import {
    listStyleOptions,
    colorThemeOptions,
    colorThemeDefault,
    blockSpaceOptions,
    blockSpaceDefault,
    iconSizeOptions,
    iconSizeDefault
} from '../../shared/constants';
import {
    LeafIcon,
    CardsIcon,
    OrganizationIcon,
    MealIcon,
    CheckBoldIcon
} from '../Icon/Icon';
import Text from '../Text/Text';
import Title from '../Title/Title';
import ListItem from '../ListItem/ListItem';
import List from './List';

const colorThemeLabel = 'Color theme';
const bulletSizeLabel = 'Bullet size';

const sampleWithIcons = [
    {
        icon: <LeafIcon />,
        main: 'First content',
        details: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
    },
    {
        icon: <CardsIcon />,
        main: 'Second content',
        details: 'Lorem ipsum dolor sit amet.'
    },
    {
        icon: <OrganizationIcon />,
        main: 'Third content',
        details: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, consectetur adipiscing elit.'
    },
    {
        icon: <MealIcon />,
        main: 'Fourth content',
        details: 'Lorem ipsum dolor sit amet consectetur.'
    }
];

const sampleBasic = [
    'Lorem ipsum dolor sit amet, consectetur adipiscing',
    'Lorem ipsum dolor sit amet',
    'Lorem ipsum dolor sit amet consectetur',
    'Lorem ipsum dolor sit adipiscing',
];

storiesOf('List', module).addDecorator(withKnobs)
.add('List with icons', () => (
    <List
        listStyle={listStyleOptions.icon}
        marginLateral={select('Lateral margins', blockSpaceOptions, blockSpaceDefault)}
        marginTop={select('Margin top', blockSpaceOptions, blockSpaceDefault)}
        marginBottom={select('Margin bottom', blockSpaceOptions, blockSpaceDefault)}
    >
        {sampleWithIcons.map((element, index) =>
            <ListItem
                key={index}
                href="#"
                bulletStyle={listStyleOptions.icon}
                bulletSize={select(bulletSizeLabel, iconSizeOptions, iconSizeDefault)}
            >
                {element.icon}
    
                <Title>
                    {element.main}
                </Title>
    
                <Text>
                    {element.details}
                </Text>
            </ListItem>
        )}
    </List>
))
.add('List with check', () => (
    <List
        listStyle={listStyleOptions.icon}
    >
        {sampleBasic.map((element, index) =>
            <ListItem
                key={index}
                bulletStyle={listStyleOptions.icon}
                bulletSize={select(bulletSizeLabel, iconSizeOptions, iconSizeDefault)}
            >
                <CheckBoldIcon />

                <Text>
                    {element}
                </Text>
            </ListItem>
        )}
    </List>
))
.add('List with dash', () => (
    <List>
        {sampleBasic.map((element, index) =>
            <ListItem
                key={index}
                bulletSize={select(bulletSizeLabel, iconSizeOptions, iconSizeDefault)}
                colorTheme={radios(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
            >
                <Text>
                    {element}
                </Text>
            </ListItem>
        )}
    </List>
))
.add('List with numbers', () => (
    <List
        listStyle={listStyleOptions.number}
    >
        {sampleBasic.map((element, index) =>
            <ListItem
                key={index}
                bulletStyle={listStyleOptions.number}
                bulletSize={select(bulletSizeLabel, iconSizeOptions, iconSizeDefault)}
                colorTheme={radios(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
            >
                <Text>
                    {element}
                </Text>
            </ListItem>
        )}
    </List>
));