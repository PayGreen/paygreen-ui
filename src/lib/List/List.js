import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
    listStyleOptions,
    listStyleDefault,
    iconSizeOptions,
    iconSizeDefault,
    colorThemeOptions,
    colorThemeDefault,
    spaceOptions,
    spaceDefault
} from '../../shared/constants';
import { ListBase } from './style';
import { dashed } from './style/dashed';

class List extends PureComponent {
    render() {
        const dashedTop = <span className="dashed top">{dashed}</span>;
        const dashedBottom = <span className="dashed bottom">{dashed}</span>;

        return <ListBase
            as={this.props.listStyle === listStyleOptions.number ? 'ol' : 'ul'}
            {...this.props}
        >
            {React.Children.map(this.props.children, (child, index) =>
                child ?
                    <li key={index}>
                        {this.props.hasDashed && index%2 ? dashedTop : null}

                        {React.cloneElement(child, { number: index + 1 })}

                        {this.props.hasDashed && index%2
                        && (index + 1 < this.props.children.length) ?
                            dashedBottom :
                            null
                        }
                    </li>
                    : null
            )}
        </ListBase>;
    }
}

List.propTypes = {
    listStyle: PropTypes.oneOf(Object.values(listStyleOptions)),
    hasDashed: PropTypes.bool,
    bulletSize: PropTypes.oneOf(Object.values(iconSizeOptions)),
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    marginLateral: PropTypes.oneOf(Object.values(spaceOptions)),
    marginTop: PropTypes.oneOf(Object.values(spaceOptions)),
    marginBottom: PropTypes.oneOf(Object.values(spaceOptions)),
};

List.defaultProps = {
    listStyle: listStyleDefault,
    hasDashed: false,
    bulletSize: iconSizeDefault,
    colorTheme: colorThemeDefault,
    marginLateral: spaceDefault,
    marginTop: spaceDefault,
    marginBottom: spaceDefault,
};

export default List;