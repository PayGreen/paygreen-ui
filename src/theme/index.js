import { ThemeDefault } from './theme.default';
import { ThemeDark } from './theme.dark';

export { ThemeDefault, ThemeDark };
