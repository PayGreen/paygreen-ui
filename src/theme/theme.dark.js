import patternDark from './assets/pattern-dark.png';
import { baseDark, gradients } from './colors';
import { ThemeBase } from './theme.base';

const ThemeDark = {
    ...ThemeBase,
    name: 'Dark',
    color: {
        primary: {
            ...baseDark.green,
            ...gradients.green,
        },
        secondary: {
            ...baseDark.blue,
            ...gradients.blue,
        },
        tertiary: {
            ...baseDark.orange,
            ...gradients.orange,
        },
        quaternary: {
            ...baseDark.pink,
            ...gradients.pink,
        },
        quinary: {
            ...baseDark.purple,
            ...gradients.purple,
        },
    },
    status: {
        success: {
            ...baseDark.green,
        },
        default: {
            ...baseDark.blue,
        },
        warning: {
            ...baseDark.orange,
        },
        danger: {
            ...baseDark.red,
        },
    },
    wab: {
        white00: '#202020',
        white10: '#2c2c2c',
        white20: '#383838',

        grey10: '#494949',
        grey20: '#5b5b5b',
        grey30: '#7f7f7f',
        grey40: '#a2a2a2',
        grey50: '#cfcfcf',
        grey60: '#eeeeee',

        black00: '#ffffff',
    },
    pattern: patternDark,
    logoColor: '#ffffff',
};

export { ThemeDark };
