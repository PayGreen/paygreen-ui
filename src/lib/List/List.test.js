import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import List from './List';

it('renders without crashing', () => {
    const component = TestRenderer.create(
        <List theme={ThemeDefault}>
            Something
        </List>
    );
    expect(component.toJSON()).toMatchSnapshot();
});