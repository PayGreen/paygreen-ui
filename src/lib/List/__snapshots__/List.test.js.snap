// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ul
  className="style__ListBase-sc-1a6882z-0 lcmBjs"
>
  <li>
    <span
      className="style__ListItemBase-sc-176tc5j-0 fBMRKh"
      type={null}
    >
      <span
        className="bullet"
      />
      <p
        className="style__TextBase-sc-11ixwvd-0 gQgxOT"
      >
        Something
      </p>
    </span>
  </li>
  <li>
    <span
      className="style__ListItemBase-sc-176tc5j-0 fBMRKh"
      type={null}
    >
      <span
        className="bullet"
      />
      <p
        className="style__TextBase-sc-11ixwvd-0 gQgxOT"
      >
        Something else
      </p>
    </span>
  </li>
</ul>
`;
